ompi/mca/hook/comm_method/hook_comm_method_fns.c

```diff
@@ -874,8 +873,375 @@ ompi_report_comm_methods(int called_from_location) // 1 = from init, 2 = from fi
+        *affstring_cursz += (amount + 16); \
+        *affstring = realloc(*affstring, *affstring_cursz); \
+    } \
+} while (0)
+
+/*
+ *  If the old --report-bindings output would have been this:
+ *            [BB/../../..][../../../..]
+ *            [../BB/../..][../../../..]
+ *            [../../../..][BB/../../..]
+ *            [../../../..][../BB/../..]
+ *  this function returns a string
+ *            [aa/bb/../..][cc/dd/../..]
+ *
+ *  This function decides how many ranks' bitmasks to process based on
+ *  1. avoiding overlap, and
+ *  2. only using 52 letters a-zA-Z
+ *  The *position argument should come in as 0 for the first call, and
+ *  it gets updated to point to the next host needing to be processed.
+ *
+ *  Each call to this function adds another line to an existing *affstring
+ *  (realloc()ing if needed).  The second-and-later lines get a carriage
+ *  return in front of them.  The last line does not end with a carriage
+ *  return.
+ */
+static void
+sprint_bitmaps(char **affstring, int *affstring_cursz,
+    int indentation,
+    hwloc_cpuset_t *cpus_forrank, int nranks,
+    int *position)
+{
+    int i, r;
+    int nchildren_done[32], depth, some_cores_printed_under_containing_obj;
+    int nextchar;
+    hwloc_obj_t obj;
+    hwloc_bitmap_t cpus;
+    char *key = "abcdefghijklmnopqrstuvwxyzABCDEGHIJKLMNOPQRSTUVWXYZ";
+    int rankA, rankB;
+
+/* pre-computing how many ranks to use
+ */
+    cpus = hwloc_bitmap_alloc();
+    r = *position;
+    hwloc_bitmap_copy(cpus, cpus_forrank[r]);
+    while (r < nranks-1) {
+        if (!hwloc_bitmap_intersects(cpus, cpus_forrank[r+1]) &&
+            (r - *position + 1) < (int)strlen(key))
+        {
+            ++r;
+            hwloc_bitmap_or(cpus, cpus, cpus_forrank[r]);
+        } else {
+            break;
+        }
+    }
+    hwloc_bitmap_free(cpus);
+/* update position, save a rankA and rankB to note the range we'll use */
+    rankA = *position;
+    rankB = r;
+    *position = *position + (rankB - rankA + 1);
+
+    nextchar = strlen(*affstring);
+    LENGTHEN(affstring, nextchar, affstring_cursz, indentation + 1);
+    if (rankA != 0) {
+        (*affstring)[nextchar++] = '\n';
+        (*affstring)[nextchar] = 0;
+        for (i=0; i<indentation; ++i) {
+            (*affstring)[nextchar++] = ' ';
+        }
+        (*affstring)[nextchar] = 0;
+    }
+
+/* for bookkeeping:
+ * let the top of the stack be the obj we're processing, and keep
+ * nchildren_done in the stack for each obj
+ */
+    obj = hwloc_get_root_obj(opal_hwloc_topology);
+    depth = 0;
+    nchildren_done[depth] = 0;
+    while (obj) {
+        LENGTHEN(affstring, nextchar, affstring_cursz, 2);
+        if (obj->type == HWLOC_OBJ_PACKAGE) {
+            (*affstring)[nextchar++] = '[';
+            (*affstring)[nextchar] = 0;
+            some_cores_printed_under_containing_obj = 0;
+        }
+        if (obj->type == HWLOC_OBJ_CORE && some_cores_printed_under_containing_obj) {
```

Comment:
`some_cores_printed_under_containing_obj` could be uninitialized here - though I think the root object's type should be `HWLOC_OBJ_PACKAGE`....I'm betting there is a compiler warning anyway since it isn't that smart.

Reply:
(pushing another commit, intend to squash once confirmed it's what we want)

I agree.  I forget what's allowable as far as degenerate cases.  I think the trees have to have a MACHINE at the top and PU at the bottom but I doubt packages have to exist.  So initializing at the top of the function would be good for the degenerate cases.